models.py

```diff
@@ -906,31 +906,33 @@ class Attendance(ndb.Model):
   jdata = ndb.JsonProperty()
 
   # 'c_id' is a class id.
-  # 'submitted_date' is the date attendance was taken.
-  # 'jdata' is a dictionary:
-  #   { 'absent': [email1, email2, ...],
-  #     'present': [email1, email2, ...],
-  #     'submitted_by': email,
-  #     'note': string }
-  @classmethod
-  def attendance_key(cls, institution, session, c_id, submitted_date):
+  # 'jdata' is a dictionary whose keys are dates which map to objects
+  # representing attendance taken that date.
+  # {date1: { 'absent': [email1, email2, ...],
+  #           'present': [email1, email2, ...],
+  #           'submitted_by': email,
+  #           'submitted_date': date1,
+  #           'note': string },
+  #  date2: { . . . }
+  # }
```

Comment:
Modified attendance object. At first, I stored attendance by class and date, but when I went to pull out the data, I realized if I didn't have a date, I'd have to search all dates in a semester. That took too long.

So instead, attendance is stored by class. Each class holds every attendance taken for that class in that session/institution. This should be fine, because there are only 12-14 (or 24-28) meetings in a semester since each selective meets once (or twice) a week.